StaticTest.test_unicode fails on codeanywhere

```
____________________________________________________________________________________________ StaticTest.test_unicode _____________________________________________________________________________________________
                                                                                                                                                                                                                  
self = <cherrypy.test.test_static.StaticTest testMethod=test_unicode>                                                                                                                                             
                                                                                                                                                                                                                  
    @pytest.mark.xfail(py27_on_windows, reason="#1544")                                                                                                                                                           
    def test_unicode(self):                                                                                                                                                                                       
>       with self.unicode_file():                                                                                                                                                                                 
                                                                                                                                                                                                                  
cherrypy/test/test_static.py:379:                                                                                                                                                                                 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python2.7/contextlib.py:17: in __enter__                                                                                                                                                                 
    return self.gen.next()                                                                                                                                                                                        
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
                                                                                                                                                                                                                  
    @staticmethod                                                                                                                                                                                                 
    @contextlib.contextmanager                                                                                                                                                                                    
    def unicode_file():                                                                                                                                                                                           
        filename = ntou('Слава Україні.html', 'utf-8')                                                                                                                                                            
        filepath = os.path.join(curdir, 'static', filename)                                                                                                                                                       
>       with io.open(filepath, 'w', encoding='utf-8') as strm:                                                                                                                                                    
E       UnicodeEncodeError: 'ascii' codec can't encode characters in position 43-47: ordinal not in range(128)                                                                                                    
                                                                                                                                                                                                                  
cherrypy/test/test_static.py:366: UnicodeEncodeError                                                                                                                                                              
===================================================================== 1 failed, 252 passed, 4 skipped, 2 xfailed, 5 xpassed in 80.13 seconds =====================================================================
ERROR: InvocationError: '/home/cabox/workspace/.tox/python/bin/pytest'                                                                                                                                            
____________________________________________________________________________________________________ summary _____________________________________________________________________________________________________
ERROR:   python: commands failed
```

Running tox on codeanywhere causes this test to fail.
* **I'm submitting a ...**
[x ] bug report
[ ] feature request
[ ] question about the decisions made in the repository

--- a/cherrypy/test/test_static.py
+++ b/cherrypy/test/test_static.py
@@ -17,6 +17,17 @@ from cherrypy._cpcompat import (
 from cherrypy.test import helper
 
 
+@pytest.fixture
+def unicode_filesystem(tmpdir):
+    filename = tmpdir / ntou('☃', 'utf-8')
+    tmpl = "File system encoding ({encoding}) cannot support unicode filenames"
+    msg = tmpl.format(encoding=sys.getfilesystemencoding())
+    try:
+        io.open(filename, 'w')
+    except UnicodeEncodeError:
+        pytest.skip(reason=msg)
+
+
 curdir = os.path.join(os.getcwd(), os.path.dirname(__file__))
 has_space_filepath = os.path.join(curdir, 'static', 'has space.html')
 bigfile_filepath = os.path.join(curdir, 'static', 'bigfile.log')
@@ -375,7 +386,7 @@ class StaticTest(helper.CPWebCase):
         sys.version_info < (3,)
     )
     @pytest.mark.xfail(py27_on_windows, reason="#1544")
-    def test_unicode(self):
+    def test_unicode(self, unicode_filesystem):
         with self.unicode_file():
             url = ntou('/static/Слава Україні.html', 'utf-8')
             # quote function requires str

--- a/cherrypy/test/test_static.py
+++ b/cherrypy/test/test_static.py
@@ -4,10 +4,13 @@ import io
 import os
 import sys
 import platform
+import tempfile
 
+import six
 from six.moves import urllib
 
 import pytest
+import py.path
 
 import cherrypy
 from cherrypy.lib import static
@@ -23,9 +26,9 @@ def unicode_filesystem(tmpdir):
     tmpl = "File system encoding ({encoding}) cannot support unicode filenames"
     msg = tmpl.format(encoding=sys.getfilesystemencoding())
     try:
-        io.open(filename, 'w')
+        io.open(six.text_type(filename), 'w')
     except UnicodeEncodeError:
-        pytest.skip(reason=msg)
+        pytest.skip(msg)
 
 
 curdir = os.path.join(os.getcwd(), os.path.dirname(__file__))
@@ -386,7 +389,8 @@ class StaticTest(helper.CPWebCase):
         sys.version_info < (3,)
     )
     @pytest.mark.xfail(py27_on_windows, reason="#1544")
-    def test_unicode(self, unicode_filesystem):
+    def test_unicode(self):
+        self.ensure_unicode_filesystem()
         with self.unicode_file():
             url = ntou('/static/Слава Україні.html', 'utf-8')
             # quote function requires str
@@ -396,6 +400,17 @@ class StaticTest(helper.CPWebCase):
 
             expected = ntou('Героям Слава!', 'utf-8')
             self.assertInBody(expected)
+    
+    def ensure_unicode_filesystem(self):
+        """
+        TODO: replace with simply pytest fixtures once webtest.TestCase
+        no longer implies unittest.
+        """
+        tmpdir = py.path.local(tempfile.mkdtemp())
+        try:
+            unicode_filesystem(tmpdir)
+        finally:
+            tmpdir.remove()
 
 
 def error_page_404(status, message, traceback, version):

--- a/cherrypy/test/test_static.py
+++ b/cherrypy/test/test_static.py
@@ -6,7 +6,7 @@ import sys
 import platform
 import tempfile
 
-import six
+from six import text_type as str
 from six.moves import urllib
 
 import pytest
@@ -26,7 +26,7 @@ def unicode_filesystem(tmpdir):
     tmpl = "File system encoding ({encoding}) cannot support unicode filenames"
     msg = tmpl.format(encoding=sys.getfilesystemencoding())
     try:
-        io.open(six.text_type(filename), 'w')
+        io.open(str(filename), 'w')
     except UnicodeEncodeError:
         pytest.skip(msg)
 
@@ -400,7 +400,7 @@ class StaticTest(helper.CPWebCase):
 
             expected = ntou('Героям Слава!', 'utf-8')
             self.assertInBody(expected)
-    
+
     def ensure_unicode_filesystem(self):
         """
         TODO: replace with simply pytest fixtures once webtest.TestCase

--- a/cherrypy/test/test_static.py
+++ b/cherrypy/test/test_static.py
@@ -26,7 +26,7 @@ def unicode_filesystem(tmpdir):
     tmpl = "File system encoding ({encoding}) cannot support unicode filenames"
     msg = tmpl.format(encoding=sys.getfilesystemencoding())
     try:
-        io.open(str(filename), 'w')
+        io.open(str(filename), 'w').close()
     except UnicodeEncodeError:
         pytest.skip(msg)
 

--- a/cherrypy/test/test_static.py
+++ b/cherrypy/test/test_static.py
@@ -31,6 +31,18 @@ def unicode_filesystem(tmpdir):
         pytest.skip(msg)
 
 
+def ensure_unicode_filesystem():
+    """
+    TODO: replace with simply pytest fixtures once webtest.TestCase
+    no longer implies unittest.
+    """
+    tmpdir = py.path.local(tempfile.mkdtemp())
+    try:
+        unicode_filesystem(tmpdir)
+    finally:
+        tmpdir.remove()
+
+
 curdir = os.path.join(os.getcwd(), os.path.dirname(__file__))
 has_space_filepath = os.path.join(curdir, 'static', 'has space.html')
 bigfile_filepath = os.path.join(curdir, 'static', 'bigfile.log')
@@ -390,7 +402,7 @@ class StaticTest(helper.CPWebCase):
     )
     @pytest.mark.xfail(py27_on_windows, reason="#1544")
     def test_unicode(self):
-        self.ensure_unicode_filesystem()
+        ensure_unicode_filesystem()
         with self.unicode_file():
             url = ntou('/static/Слава Україні.html', 'utf-8')
             # quote function requires str
@@ -400,17 +412,6 @@ class StaticTest(helper.CPWebCase):
 
             expected = ntou('Героям Слава!', 'utf-8')
             self.assertInBody(expected)
-
-    def ensure_unicode_filesystem(self):
-        """
-        TODO: replace with simply pytest fixtures once webtest.TestCase
-        no longer implies unittest.
-        """
-        tmpdir = py.path.local(tempfile.mkdtemp())
-        try:
-            unicode_filesystem(tmpdir)
-        finally:
-            tmpdir.remove()
 
 
 def error_page_404(status, message, traceback, version):

